Release Notes/Release Notes v2.9.0.md

```diff
@@ -1,3 +1,30 @@
 # .NET Driver Version 2.9.0 Release Notes
 
-This file is a placeholder.
\ No newline at end of file
+The main new features in 2.9.0 are:
+
+* Sharded transactions using the same API as replica set transactions
+* Sharded transactions pinning to a single mongos router
+* Convenient API for transactions
+* Support for message compression
+* SRV polling for mongodb+srv connection scheme
+* Retryable reads on by default
+* Retryable writes on by default
+* Update specification using an aggregation framework pipeline
+* SCRAM-SHA authentication caching
+
```

Comment:
I combined the beta features into this list, minimally rewording things. What else do we want on this list?

Reply:
Other things Java driver has in release notes:

* Connections to the replica set primary are no longer closed after a step-down, allowing in progress read operations to complete.
* New aggregate helper methods support running database-level aggregations.
* Aggregate helper methods now support the $merge pipeline stage, and Aggregates.merge() builder methods support creation of the new pipeline stage.
* Change stream helpers now support the startAfter option.
* Index creation helpers now support wildcard indexes.

Reply:
Added!